core/endorser/endorser.go

```diff
@@ -500,3 +507,48 @@ func CreateCCEventBytes(ccevent *pb.ChaincodeEvent) ([]byte, error) {
 func decorateLogger(logger *flogging.FabricLogger, txParams *ccprovider.TransactionParams) *flogging.FabricLogger {
 	return logger.With("channel", txParams.ChannelID, "txID", shorttxid(txParams.TxID))
 }
+
+// Using the simulation results, build the ChaincodeInterest structure that the client can pass to the discovery service
+// to get the correct endorsement policy for the chaincode(s) and any collections encountered.
+func buildChaincodeInterest(simResult *ledger.TxSimulationResults) (*pb.ChaincodeInterest, error) {
+	ccInterest := &pb.ChaincodeInterest{}
+
+	chaincodes := map[string]struct{}{}
+	if simResult.PvtSimulationResults != nil {
+		// prototype - list collections that are read from (in order to populate NoPublicReads in ChaincodeInterest)
+		for _, prws := range simResult.PvtSimulationResults.GetNsPvtRwset() {
+			for _, cprws := range prws.GetCollectionPvtRwset() {
+				ccInterest.Chaincodes = append(ccInterest.Chaincodes, &pb.ChaincodeCall{
+					Name:            prws.GetNamespace(),
+					CollectionNames: []string{cprws.GetCollectionName()},
+					NoPrivateReads:  !simResult.PrivateReads.Read(prws.Namespace, cprws.CollectionName),
+				})
+				chaincodes[prws.GetNamespace()] = struct{}{}
+			}
+		}
+	}
+	// if there are any chaincodes in the public RWset that are not in the private set, add them to the interest
+	if simResult.PubSimulationResults != nil {
+		for _, nsrws := range simResult.PubSimulationResults.GetNsRwset() {
+			if nsrws.GetNamespace() == "_lifecycle" {
+				continue
```

Comment:
At other places we avoided this hardcoding via [this function](https://github.com/hyperledger/fabric/blob/9a922fd9c78c01d185bc2b45fb98012a37f7bfb9/core/chaincode/lifecycle/deployedcc_infoprovider.go#L52). see if you could use this here easily. Otherwise, can introduce a function in that package that returns these names - which would cover the name for legacy lifecycle name as well (i.e., `lscc`).